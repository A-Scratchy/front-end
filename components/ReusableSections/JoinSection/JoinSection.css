.JoinSection {
  align-items: center;
  max-width: 900px;
  text-align: center;
}

.form {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  padding: 1rem 0;
}

.form > input {
  width: 300px;
}
