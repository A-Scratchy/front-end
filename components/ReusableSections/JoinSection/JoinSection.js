import React from 'react';
import Container from 'components/Container/Container';
import OutboundLink from 'components/OutboundLink/OutboundLink';
import LinkButton from 'components/LinkButton/LinkButton';
import styles from './JoinSection.css';

const JoinSection = () => (
  <Container theme="white">
    <h3>Join Our Thriving Community</h3>

    <p className={styles.justifyAlign}>
      Are you ready to begin your journey towards a career in software development? Get the support
      you need by joining our members-only Slack community!
    </p>

    <LinkButton href="/join" theme="secondary" className={styles.verticalSpacing}>
      Register Now
    </LinkButton>

    <p>Slack is a community based collaboration tool where all the magic happens!</p>

    <OutboundLink href="https://slack.com/" analyticsEventLabel="Learn More About Slack">
      Learn more
    </OutboundLink>
  </Container>
);

export default JoinSection;
