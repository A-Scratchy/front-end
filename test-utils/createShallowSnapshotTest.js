/* eslint-env jest */
import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers or larger components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
