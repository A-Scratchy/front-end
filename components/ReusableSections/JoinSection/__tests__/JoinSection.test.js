import React from 'react';
import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';
import JoinSection from '../JoinSection';

describe('JoinSection', () => {
  it('it should render properly with no props', () => createShallowSnapshotTest(<JoinSection />));
});
