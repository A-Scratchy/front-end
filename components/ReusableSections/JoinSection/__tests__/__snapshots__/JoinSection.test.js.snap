// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinSection should render when not logged in 1`] = `
<div
  className="Container white"
  data-testid="Join Section"
>
  <div
    className="content"
  >
    <div
      className="headingContainer"
    >
      <span
        className="anchorSpan anchorMargin"
        id="join-our-thriving-community"
      />
      <div
        className="hashLinkContainer"
      >
        <a
          data-testid="Hash Link"
          href="#join-our-thriving-community"
        >
          <span
            className="ScreenReaderOnly"
          >
            Scroll Link
          </span>
          <svg
            className="icon"
          />
        </a>
        <h3
          className="white headingTextWithLinkIconOffset Heading"
        >
          Join Our Thriving Community
        </h3>
      </div>
    </div>
    <p
      className="justifyAlign"
    >
      Are you ready to begin your journey towards a career in software development? Get the support you need by joining our members-only Slack community!
    </p>
    <a
      className="Button verticalSpacing secondary"
      href="/join"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Register Now
    </a>
    <p>
      Slack is a community based collaboration tool where all the magic happens!
    </p>
    <a
      href="https://get.slack.help/hc/en-us/categories/360000049043-Getting-Started"
      onClick={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        className="ScreenReaderOnly"
      >
        Opens in new window
      </span>
      Never heard of Slack before?
      <svg
        className="externalLinkIcon"
      />
    </a>
  </div>
</div>
`;
